pyproject.toml

```diff
@@ -66,7 +66,7 @@ exclude_lines = ["pragma: no cover", "if TYPE_CHECKING"]
 
 [tool.pytest.ini_options]
 filterwarnings = [
-    "error:::marray.*",
+    "error",
```

Comment:
@keewis Any reason not to do this? There were `UserWarning`s that weren't being raised as errors before.

Reply:
the idea was that anything raised within `marray` should be caught using `pytest.warns`, while warnings outside of `marray` are usually out of control so we can't fix it (like a deprecation warning raised in a dependency – which `marray` doesn't really have anymore).

I guess by making all warnings an error we'd need to explicitly ignore any warning that we can't do anything about.

Reply:
I see. I might switch it back, but it hasn't been too bad to ignore them on a case by case basis. We already needed to ignore the usual NumPy warnings (invalid, divide, over) separately because I'm pretty confident that those won't be due to a real mistake.